### PR TITLE
Use a single instance of `rich.console.Console` throughout the codebase

### LINKED ISSUE #12824: Rich progress bar doesn't play nicely with other prints
## 🐛 Bug

Some internal printing mechanisms don't play nicely with RichProgressBar.
This can result in the progress bar getting interleaved into the printed data.

For example, this affects the pretty table of metrics that is printed after `Trainer.test`ing your model:
```
Testing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1/1 0:00:00 • 0:00:00 0.00it/s  ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│         test_loss         │    -4.4067792892456055    │
└───────────────────────────┴───────────────────────────┘
```

A similar issue can also come up, when using `RichModelSummary.summarize`.

### To Reproduce

Take the `BoringModel` sample code and add 
```
callbacks=[pl.callbacks.RichProgressBar()]
```
to the `Trainer`.

### Expected behavior

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│         test_loss         │    -4.4067792892456055    │
└───────────────────────────┴───────────────────────────┘
Testing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1/1 0:00:00 • 0:00:00 0.00it/s
```

### Environment

```
* CUDA:
	- GPU:
		- NVIDIA GeForce GTX 1050
	- available:         True
	- version:           11.3
* Packages:
	- numpy:             1.22.3
	- pyTorch_debug:     False
	- pyTorch_version:   1.11.0+cu113
	- pytorch-lightning: 1.6.0
	- tqdm:              4.63.2
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         
	- python:            3.10.4
	- version:           #1 SMP PREEMPT Mon Mar 28 09:16:36 UTC 2022
```

## Fix

Instead of creating a new `Console` and printing to it, `RichModelSummary.summarize` and `EvaluationLoop._print_results` should <s>check if `trainer` has a `progress_bar_callback` of type `RichProgressBar` and if so, it should use `trainer.progress_bar_callback.progress.console`.</s>

Edit: Actually a better solution might be using `rich.get_console()` instead of creating new instances of `rich.console.Console()` everywhere and if you need to change the `kwargs` (because user passed `console_kwargs` in `RichProgressBar`) use `rich.reconfigure(**console_kwargs)`.

cc @kaushikb11 @rohitgr7

### PR DESCRIPTION
**Pull request by otaj**
## What does this PR do?

Use a single instance of rich Console throughout the codebase, so that everything plays nicely together.

Fixes #12824 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @kaushikb11 @rohitgr7

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/loops/dataloader/evaluation_loop.py [hunk 1]
```diff
@@ -358,16 +358,18 @@ def _print_results(results: List[_OUT_DICT], stage: str, file: Optional[IO[str]]
...
                 table = Table(*columns)
                 for metric, row in zip(metrics, table_rows):
                     row.insert(0, metric)
                     table.add_row(*row)
-                console.print(table)
+
+                console = get_console()
+                with console.capture() as capture:
+                    console.print(table)
+                print(capture.get(), end="", file=file)
```
Comment (carmocca) [inline pytorch_lightning/loops/dataloader/evaluation_loop.py:372]:
```suggestion
                console = get_console()
                reconfigure(console, file=file)
                console.print(table)
                # TODO: remove file config
```

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 2]
```diff
@@ -278,7 +279,8 @@ def enable(self) -> None:
     def _init_progress(self, trainer):
         if self.is_enabled and (self.progress is None or self._progress_stopped):
             self._reset_progress_bar_ids()
-            self._console = Console(**self._console_kwargs)
+            reconfigure(**self._console_kwargs)
+            self._console = get_console()
```
[the remark this answers is not in the document]
Comment (otaj) [reply pytorch_lightning/callbacks/progress/rich_progress.py:283]: The documentation of rich is just weird about this, but from what I see in code, this is not the way how to invoke it

**Review by kaushikb11 — APPROVED** [review]
Awesome!

**Review by carmocca — APPROVED** [review]

File: tests/trainer/logging_/test_eval_loop_logging.py [hunk 3]
```diff
@@ -950,7 +957,42 @@ def test_native_print_results_encodings(monkeypatch, encoding):
...
+    console = get_console()
+    with console.capture() as capture:
+        EvaluationLoop._print_results(*inputs)
     expected = expected[1:]  # remove the initial line break from the """ string
-    assert out.getvalue() == expected.lstrip()
+    assert capture.get() == expected.lstrip()
+
+
+@RunIf(rich=True, skip_windows=True)
+def test_rich_print_results_with_progress_bar(tmpdir):
```
[the remark this answers is not in the document]
Comment (otaj) [reply tests/trainer/logging_/test_eval_loop_logging.py]: Done
